app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.js

```diff
@@ -211,6 +211,106 @@ window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {};
+        var resultCount = element.querySelector('#' + id).textContent
+
+        if (!resultCount) {
+          return null
+        }
+
+        resultCount = resultCount.replace(',', '')
+        resultCount = resultCount.split(' ')[0]
+
+        return parseInt(resultCount)
+      },
+
+      populateEcommerceSchema: function(data) {
+        var element = data.element
+        var resultsId = data.resultsId
+        var isClickEvent = data.event !== undefined ? true : false
+        var isSearchResult = element.getAttribute('data-search-query') !== null ? true : false
+
+        var ecommerceSchema = new window.GOVUK.analyticsGa4.Schemas().ecommerceSchema()
+        var PIIRemover = new GOVUK.analyticsGa4.PIIRemover()
+        var DEFAULT_LIST_TITLE
+
+        if(isSearchResult) {
+            // Limiting to 100 characters to avoid noise from extra long search queries and to stop the size of the payload going over 8k limit.
+            var searchQuery = PIIRemover.stripPII(element.getAttribute('data-search-query')).substring(0, 100).toLowerCase()
+            var variant = element.getAttribute('data-ecommerce-variant') || undefined
+            DEFAULT_LIST_TITLE = "Site search results"
+        }
+        else {
+          DEFAULT_LIST_TITLE = "Smart answer results"
+        }
+        
+        var items = element.querySelectorAll('[data-ecommerce-path]')
+        var listTitle = element.getAttribute('data-list-title') || DEFAULT_LIST_TITLE
+        var startPosition = parseInt(element.getAttribute('data-ecommerce-start-index'), 10)
+
+        ecommerceSchema.event = 'search_results'
+        ecommerceSchema.search_results.event_name = isClickEvent ? 'select_item' : 'view_item_list'
+        ecommerceSchema.search_results.results = window.GOVUK.analyticsGa4.core.ecommerceHelperFunctions.getResultCount(element, resultsId)
+        ecommerceSchema.search_results.term = searchQuery
+        ecommerceSchema.search_results.sort = variant
+        ecommerceSchema.search_results.ecommerce.items = []
+
+        if(isClickEvent) {
+            data.event.preventDefault()
+            var target = data.event.target
+            ecommerceSchema.search_results.ecommerce.items.push({
+                item_id: target.getAttribute('data-ecommerce-path'),
+                item_name: target.textContent,
+                item_list_name: listTitle,
+                index: window.GOVUK.analyticsGa4.core.ecommerceHelperFunctions.getIndex({
+                  element: target,
+                  startPosition: startPosition
+                })
+            })
+
+            ecommerceSchema.event_data = {
+                external: GOVUK.analyticsGa4.core.trackFunctions.isExternalLink(target.getAttribute('data-ecommerce-path')) ? 'true' : 'false'
+            }
+        }
+        else{
+            for (var i = 0; i < items.length; i++) {
```

Comment:
`items` is defined on line 262 and doesn't seem to be used until this point, could it be brought inside this `else` statement?